spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -442,6 +450,33 @@ protected void invoke(Object records, @Nullable Acknowledgment acknowledgment, @
 		}
 	}
 
+	protected void invoke(Object records, @Nullable ShareAcknowledgment acknowledgment, @Nullable ShareConsumer<?, ?> consumer,
```

Comment:
Take a look, please, if existing `invoke()` could be reworked to have `@Nullable Object acknowledgment, @Nullable Object consumer` instead.